src/PhpSpec/Matcher/ThrowMatcher.php

```diff
@@ -121,11 +121,25 @@ public function verifyPositive($callable, array $arguments, $exception = null)
         }
 
         if (!$exceptionThrown instanceof $exception) {
+            $format = 'Expected exception of class %s, but got %s.';
+
+            // Show full error message if the thrown exception is an instance of Error.
+            // See https://github.com/phpspec/phpspec/issues/908
+            if ($exceptionThrown instanceof \Error) {
+                // TODO: Can we can use $this->presenter->presentValue() rather
+                // than adding the raw Error message in quotes here? Currently
+                // not included since presentValue() truncates the string to 25
+                // characters but we want to show the full Error message for
+                // debugging purposes.
+                $format = 'Expected exception of class %s, but got %s with the message: "%s"';
+            }
+
             throw new FailureException(
                 sprintf(
-                    'Expected exception of class %s, but got %s.',
+                    $format,
                     $this->presenter->presentValue($exception),
-                    $this->presenter->presentValue($exceptionThrown)
+                    $this->presenter->presentValue($exceptionThrown),
+                    $exceptionThrown->getMessage()
```

Comment:
You mean because the 4th param of `sprintf` might not be used if `$exceptionThrown` is not an instance of `\Error`?

I guess the alternative would be to build the message for `FailureException` within the conditionals and just pass it in (shown below). That does end up with a fair bit of duplication. 

```php
if (!$exceptionThrown instanceof $exception) {
    $failure_message = sprintf(
        'Expected exception of class %s, but got %s.',
        $format,
        $this->presenter->presentValue($exception),
        $this->presenter->presentValue($exceptionThrown)
    );

    if ($exceptionThrown instanceof \Error) {
        $failure_message = sprintf(
            'Expected exception of class %s, but got %s with the message: "%s"',
            $format,
            $this->presenter->presentValue($exception),
            $this->presenter->presentValue($exceptionThrown),
            $exceptionThrown->getMessage()
        );
    }

    throw new FailureException($failure_message);
}
```

An alternative might be to just decided that the error message is always useful (since it's only shown when a test fails and it's prudent to show the developer more information if we can, not less)? That would be as shown below:

```php
if (!$exceptionThrown instanceof $exception) {
    $format = 'Expected exception of class %s, but got %s with the message: "%s".';

    throw new FailureException(
        sprintf(
            $format,
            $this->presenter->presentValue($exception),
            $this->presenter->presentValue($exceptionThrown),
            $exceptionThrown->getMessage()
        )
    );
}
```